rpmio/rpmlog.cc

```diff
@@ -22,6 +23,7 @@ struct rpmlogCtx_s {
     unsigned mask;
     int nrecsPri[RPMLOG_NPRIS];
     std::vector<rpmlogRec_s> recs;
+    std::map<uint64_t, std::map<std::pair<int, std::string>, int>> seen;
```

Comment:
Something that occurred to me last evening after work: this would be better off with unordered_map for both maps, which is STL's kind of exotic name for what the industry knows as a hash map, std::map maintains things in sort order at all times. Both work but unordered is (obviously) somewhat faster when you don't need the sorted iteration etc.

Not that it'll matter much here because there'll be only a handful of items in either map for the foreseeable future. So not a review requirement (although changing is trivial because both maps dance to the same exact API), just .. maybe education :sweat_smile:

Reply:
Yeah, I thought about using an unordered map in the beginning and then opted to go for the normal map just in case.

Good I did because I might not have been able to make the unordered_map work right away. The missing hash function for pairs gives a pretty long compiler message. But I got it working now.

Reply:
Oh, interesting. It didn't really even occur to me that you'd need to define a hash for it when map didn't require any custom comparison stuff for this structure. That's something to keep in mind.

Water gets deep very suddenly in C++ - on one line you're operating on ridiculously high level and on the next one you run into an absurdly long compiler error because you failed to define some low-level detail you'd kinda expect the compiler to do for you because this is all standard datatypes. Python it is not :sweat_smile:

Reply:
Another note is that it's good to document such things in the commit message: map worked out of the box, unordered_map didn't - had it been there, I wouldn't have even brought this up.